functests/sctp/sctp_sriov.go

```diff
@@ -0,0 +1,202 @@
+	runtimeclient "sigs.k8s.io/controller-runtime/pkg/client"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/openshift-kni/cnf-features-deploy/functests/utils/client"
+	"github.com/openshift-kni/cnf-features-deploy/functests/utils/discovery"
+	"github.com/openshift-kni/cnf-features-deploy/functests/utils/execute"
+	"github.com/openshift-kni/cnf-features-deploy/functests/utils/namespaces"
+	"github.com/openshift-kni/cnf-features-deploy/functests/utils/sriov"
+)
+
+const (
+	sriovOperatorNamespace = "openshift-sriov-network-operator"
+	testNetwork            = "test-sctp-sriov-network"
+)
+
+var _ = Describe("[sriov] SCTP integration", func() {
+	sriovclient := sriovtestclient.New("")
+	discoveryFailed := false
+	var testNode string
+
+	execute.BeforeAll(func() {
+		err := namespaces.Create(TestNamespace, client.Client)
+		Expect(err).ToNot(HaveOccurred())
+
+		Eventually(func() error {
+			_, err := client.Client.ServiceAccounts(TestNamespace).Get(context.Background(), "default", metav1.GetOptions{})
+			return err
+		}, 5*time.Minute, 5*time.Second).Should(Not(HaveOccurred()))
+
+		err = namespaces.Clean(TestNamespace, "testsctp-", client.Client)
+		Expect(err).ToNot(HaveOccurred())
+
+		selector := sctpNodeSelector
+		if selector == "" {
+			selector, err = findSCTPNodeSelector()
+			Expect(err).ToNot(HaveOccurred())
+		}
+		sriovSctpNodes := discoverSRIOVNodes(sriovclient, selector)
+
+		if !discovery.Enabled() {
+			err := sriovnamespaces.Clean(sriovOperatorNamespace, TestNamespace, sriovclient)
+			Expect(err).ToNot(HaveOccurred())
+			createSRIOVNetworkPolicy(sriovclient, sriovSctpNodes.Nodes[0], sriovSctpNodes, "sctptestres")
+			sriov.WaitStable(sriovclient)
+			Eventually(func() int64 {
+				testedNode, err := client.Client.Nodes().Get(context.Background(), sriovSctpNodes.Nodes[0], metav1.GetOptions{})
+				Expect(err).ToNot(HaveOccurred())
+				resNum, _ := testedNode.Status.Allocatable["openshift.io/sctptestres"]
+				capacity, _ := resNum.AsInt64()
+				return capacity
+			}, 3*time.Minute, time.Second).Should(Equal(int64(5)))
+
+		} else {
+			err := sriovnamespaces.CleanNetworks(sriovOperatorNamespace, sriovclient)
+			Expect(err).ToNot(HaveOccurred())
+			err = sriovnamespaces.CleanPods(sriovOperatorNamespace, sriovclient)
+			Expect(err).ToNot(HaveOccurred())
+		}
+
+		node, resourceName, numVfs, sriovDevice, err := sriovdiscovery.DiscoveredResources(sriovclient,
+			sriovSctpNodes, sriovOperatorNamespace, func(policy sriovv1.SriovNetworkNodePolicy) bool {
+				if policy.Spec.DeviceType != "netdevice" {
+					return false
+				}
+				return true
+			},
+			func(node string, sriovDeviceList []*sriovv1.InterfaceExt) (*sriovv1.InterfaceExt, bool) {
+				if len(sriovDeviceList) == 0 {
+					return nil, false
+				}
+				return sriovDeviceList[0], true
+			},
+		)
+		Expect(err).ToNot(HaveOccurred())
+
+		discoveryFailed = node == "" || resourceName == "" || numVfs < 2
```

Comment:
can we move the `if` to here to check if the discovery failed?

Reply:
Nope, because it's `BeforeAll`. Whenever we add a new test, the flag check must happen in BeforeEach, otherwise only the first one will be skipped.